File: odd_even_sort/go/oddeven_sort.go
package main

import(
	"fmt"
	"math/rand"
	"time"
	)





func odd_even_sort(ulist [] int, usize int) {

	sorted := true


	for sorted == true {

		sorted = false
		for i := 0; i < usize-1; i += 2 {

			if ulist[i] > ulist[i+1] {

				ulist[i], ulist[i+1] = ulist[i+1], ulist[i]
				sorted = true
			}

		}
		for j := 1; j < usize-1; j += 2 {

			if ulist[j] > ulist[j+1] {
				ulist[j], ulist[j+1] = ulist[j+1], ulist[j]
				sorted = true

			}
		}
	}
}






func main() {

        arr_size := 100000

        unsorted_slice := make([] int, arr_size, arr_size)

        rand.Seed(time.Now().UnixNano())


        for i := 0; i < arr_size; i++ {

                unsorted_slice[i] = rand.Intn(100)

        }

        fmt.Println(unsorted_slice)

        start_time := time.Now()

        odd_even_sort(unsorted_slice, arr_size)

        duration := time.Since(start_time)

        fmt.Println(unsorted_slice)

        fmt.Println("----------------Execution Time --->", duration)





}

File: odd_even_sort/python3/oddeven_sort.py
import random
import timeit









def do_work():

        unsorted_list = []
        list_size = 10000


        for i in range(1,int(list_size)):
                unsorted_list.append(random.randint(0, 100))
        print(f'------------------------> unsorted {unsorted_list}')

        start_time = timeit.default_timer()
        odd_even_sort(unsorted_list, list_size)
        end_time = timeit.default_timer()


        print(f'------------------------> sorted_list {unsorted_list}')
        print(f'------------------------> Execution list {end_time - start_time} seconds')      






def odd_even_sort(ulist, usize):

	sorted = False

	while not sorted:

		sorted = True

		for i in range(1,usize-2,2):
			if ulist[i] > ulist[i+1]:
				ulist[i], ulist[i+1] = ulist[i+1], ulist[i]
				sorted = False

		for i  in range(0,usize-2, 2):
			if ulist[i] > ulist[i+1]:
				ulist[i], ulist[i+1] = ulist[i+1], ulist[i]
				sorted = False







if __name__ == '__main__':

	do_work()
